tools/mtouch/mtouch.cs

```diff
@@ -476,6 +477,8 @@ public static string GetAotArguments (Application app, string filename, Abi abi,
 				args.Append ("llvmonly,");
 			else if (interp)
```

Comment:
I suggest to add an assert here, `interp` should only be true of `fname == "mscorlib.dll"` is true.

Reply:
Don't you mean that `interp_full` should only be true if `name == "mscorlib.dll"`?

Reply:
No, other assemblies than `mscorlib.dll` are vaild with `--aot=interp,full` too, as it will add wrappers specific to signatures seen during AOT compilation that are necessary for the AOT<>interpreter transitions.

However, `--aot=interp` only makes sense for `mscorlib.dll`. At least that's how it's supposed to work, I should double check that on the mono side: https://github.com/mono/mono/issues/10543